### ISSUE TITLE
Unexpected obsolete diagnostics on the constraint type

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#5701]
Compile the following code:
```
    [System.Obsolete]
    public static class Crash<T> 
        where T: Crash<T>.AbstractClass 
    { 
        public abstract class AbstractClass 
        { 
            public int Id { get; set; } 
        } 
    } 
```

Unexpected:
```
(3,18): warning CS0612: 'Crash<T>' is obsolete
```

**Comment by gafter** [comment]
An obsolete class should not receive obsolete diagnostics for types appearing in its base clause or constraints.

**Comment by gafter** [comment]
Closing as a dup of #4097
